media-libs/libplacebo: Add new ebuild

haasn opened the pull request:
New project for video rendering that VLC 4 plans to use more heavily.

Incidentally, this is an optional minor dependency of VLC 3, so we could
use the system-wide version instead of the vendored version in the
existing VLC ebuilds as well.

Signed-off-by: Niklas Haas <git@haasn.xyz>

haasn commented:
cc @lu-zero

mgorny commented on media-libs/libplacebo/libplacebo-0.5.0.ebuild:
```diff
@@ -0,0 +1,45 @@
+# Copyright 1999-2018 Gentoo Foundation
```
Why two versions?

haasn replied:
No reason, other than "I had them in my own tree at the time of submitting". I'll remove the version 0.5.

mgorny commented on media-libs/libplacebo/libplacebo-0.6.0.ebuild:
```diff
@@ -0,0 +1,47 @@
+	vulkan? ( media-libs/vulkan-loader[${MULTILIB_USEDEP}] )"
+DEPEND="${RDEPEND}"
+
+DOCS=( README.md )
```
I'm sorry if you know that already but you don't have to specify `DOCS` if all installed documentation happens to match the default (which includes `README*`).

haasn replied:
Okay, I'll remove it.

mgorny commented on media-libs/libplacebo/metadata.xml:
```diff
@@ -0,0 +1,17 @@
+	<maintainer type="person">
+		<email>gentoo@haasn.xyz</email>
+		<name>Niklas Haas</name>
+	</maintainer>
```
If you'd like to maintain this via proxy-maint, please add us as well (below yourself). Otherwise, you'd need some other committer.

haasn replied:
I've added proxy-maint as well.

haasn commented:
Addressed your feedback and rebased onto master. Please let me know if you see any further issues.

Zlogene commented on media-libs/libplacebo/libplacebo-0.6.0.ebuild:
```diff
@@ -0,0 +1,45 @@
+
+EAPI=6
+
+SCM=""
```
why introduce single use varaible?

Zlogene commented on media-libs/libplacebo/libplacebo-0.6.0.ebuild:
```diff
@@ -0,0 +1,45 @@
+
+SCM=""
+if [[ "${PV}" == "9999" ]]; then
+	SCM="git-r3"
```
inherit it from there

haasn replied:
That's the way I had it originally, but I was given this feedback by user @kagami (for a different but very similar ebuild, media-libs/dav1d-9999):

```
2018-09-24 08:54:07	kagami	haasn: inherit git-r3 should be first line btw
2018-09-24 08:54:17	kagami	see man for inherit
2018-09-24 08:57:36	kagami	or see how it's done in ffmpeg ebuild (SCM variable)
...
2018-09-27 15:42:14	kagami	haasn: https://0x0.st/s3kO.ebuild
```

I can change it back to inheriting from the `if` if you still prefer it that way.

Zlogene commented on media-libs/libplacebo/libplacebo-0.6.0.ebuild:
```diff
@@ -0,0 +1,45 @@
+	SRC_URI="https://github.com/haasn/libplacebo/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+fi
+
+inherit ${SCM} meson ninja-utils multilib-minimal
```
and drop here

Zlogene commented on media-libs/libplacebo/libplacebo-9999.ebuild:
```diff
@@ -0,0 +1,45 @@
+
+EAPI=6
+
+SCM=""
```
ditto for this ebuild

a17r commented:
> Incidentally, this is an optional minor dependency of VLC 3, so we could use the system-wide version instead of the vendored version in the existing VLC ebuilds as well.

Well, vlc currently has --disable-libplacebo and I'm not aware of a bundled lib. That also means you won't introduce an unhandled dependency for vlc when this enters tree, so we're good and can add support in vlc afterwards. :)

haasn commented:
> Well, vlc currently has --disable-libplacebo and I'm not aware of a bundled lib. That also means you won't introduce an unhandled dependency for vlc when this enters tree, so we're good and can add support in vlc afterwards. :)

Oh, then that's also fine. The bundled lib would be `contrib/src/libplacebo` inside the VLC source tree. But if you pass `--disable-libplacebo` or ignore the contribs altogether, we're fine.

haasn commented:
Addressed @mgorny's feedback and reworded the commit message to reflect @a17r's comment.

mgorny commented:
Build fails with the following error:
```
../libplacebo-0.6.0/src/spirv_shaderc.c: In function ‘shaderc_init’:
../libplacebo-0.6.0/src/spirv_shaderc.c:49:13: error: ‘shaderc_optimization_level_performance’ undeclared (first use in this function); did you mean ‘shaderc_optimization_level_zero’?
             shaderc_optimization_level_performance);
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
             shaderc_optimization_level_zero
```
[media-libs:libplacebo-0.6.0:20181018-175151.log](https://github.com/gentoo/gentoo/files/2493039/media-libs.libplacebo-0.6.0.20181018-175151.log)

mgorny commented:
Yes, 4. would be best.

Coacher commented:
Newer shaderc release already prepared and waiting for Gentoo devs' action for almost two weeks:
https://github.com/gentoo/gentoo/pull/10101

haasn commented:
I've updated and rebased this PR for the release of the new libplacebo version 1.7.0. Notably, this libplacebo version drops the requirement on shaderc >= 2018.0, and is therefore compatible with the (currently in gentoo) version >= 2017.2 as well. So the major blocker to having this be merged is now removed.

a17r commented on media-libs/libplacebo/libplacebo-9999.ebuild:
```diff
@@ -0,0 +1,47 @@
+RDEPEND="glslang? ( <dev-util/glslang-7.10[${MULTILIB_USEDEP}] )
+	shaderc? ( >=media-libs/shaderc-2017.2[${MULTILIB_USEDEP}] )
+	vulkan? ( media-libs/vulkan-loader[${MULTILIB_USEDEP}] )
+	lcms? ( media-libs/lcms:2[${MULTILIB_USEDEP}] )"
```
Please keep those lines sorted lexicographically.

a17r commented on media-libs/libplacebo/metadata.xml:
```diff
@@ -0,0 +1,21 @@
+		<flag name="glslang">Use dev-libs/glslang to compile GLSL</flag>
+		<flag name="shaderc">Use media-libs/shaderc to compile GLSL</flag>
+		<flag name="vulkan">Add support for the vulkan graphics API</flag>
+		<flag name="lcms">Add support for color management via LittleCMS</flag>
```
sorting here as well

a17r commented on media-libs/libplacebo/libplacebo-9999.ebuild:
```diff
@@ -0,0 +1,47 @@
+		-D glslang=$(usex glslang enabled disabled)
+		-D shaderc=$(usex shaderc enabled disabled)
+		-D vulkan=$(usex vulkan enabled disabled)
+		-D lcms=$(usex lcms enabled disabled)
```
sorting here as well

haasn replied:
It does work. Would you be happier without the space?

a17r replied:
I'm not going to block you from adding it like that, but it seems without space is what is preferred throughout tree

haasn replied:
Changed

a17r commented on media-libs/libplacebo/metadata.xml:
```diff
@@ -0,0 +1,21 @@
+		<name>Gentoo Proxy Maintainers Project</name>
+	</maintainer>
+	<use>
+		<flag name="glslang">Use dev-libs/glslang to compile GLSL</flag>
```
This and line below: Please use the `<pkg>` tag around package atoms.

a17r commented on media-libs/libplacebo/libplacebo-1.7.0.ebuild:
```diff
@@ -0,0 +1,47 @@
+	inherit git-r3
+else
+	KEYWORDS="~amd64"
+	SRC_URI="https://code.videolan.org/videolan/libplacebo/-/archive/v${PV}/libplacebo-v${PV}.tar.gz -> ${P}.tar.gz"
```
PS: technically, that is unnecessary cosmetics (filename is vaild without modification)

haasn replied:
Oh, indeed. Changed

a17r commented on media-libs/libplacebo/libplacebo-1.7.0.ebuild:
```diff
@@ -0,0 +1,47 @@
+# Copyright 1999-2018 Gentoo Foundation
```
PS: I recommend using `repoman commit`. Among basic QA checks and adding your sign-off, it would have auto-fixed this copyright header.

haasn replied:
I don't understand, what's wrong about this copyright header. `repoman scan` reports no issues for me. (But thanks for the tip!)

a17r replied:
wrong it is not, but since signoff requirement, repoman commit will auto-fix old headers to `Gentoo Authors` instead.

mgorny commented:
I'm sorry this took so long. I see you've made a new release recently but nevertheless I'm going to merge this as it's good quality. I hope you don't get discouraged and bump it at your leisure.
